lib/ansible/modules/cloud/amazon/s3_sync.py

```diff
@@ -437,25 +446,29 @@ def upload_files(s3, bucket, filelist, params):
         }
         if params.get('permission'):
             args['ACL'] = params['permission']
-        s3.upload_file(entry['fullpath'], bucket, entry['s3_path'], ExtraArgs=args, Callback=None, Config=None)
+        try:
+            s3.upload_file(entry['fullpath'], bucket, entry['s3_path'], ExtraArgs=args, Callback=None, Config=None)
+        except botocore.exceptions.ClientError as err:
+            # caught in main()
+            raise
```

Comment:
what is the point of this though if we're just going to raise - won't it be caught at the parent even without the `try/except` block?